### PR TITLE
Fixed the new problems in kernel compilation in redhat distro

### PR DESCRIPTION
**Pull request by juhlee-microsoft**
1. Deprecated iperf. Moved to iperf3.
2. Fixed the kernel config update
3. Added a new common function found_sys_log; return 1 if found the string in the system logs.
4. Increased the Azure disk attachment operation wait time from 30 seconds to 60 seconds
5. Still has stress-ng' package installation in redhat distro

**TEST RESULTS**

[LISAv2 Test Results Summary]
Test Run On           : 10/01/2020 00:15:37
ARM Image Under Test  : RedHat : RHEL : 8.2 : latest
Override VM size Under Test  : **Standard_NC12**
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:37

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes)
-------------------------------------------------------------------------------------------------------------------------------------------
    1 POWER                POWER-HIBERNATE                                                                   PASS                35.69
      ARMImageName: RedHat RHEL 8.2 latest, OverrideVMSize: Standard_NC12, TestLocation: southcentralus, Kernel Version: 4.18.0-193.14.3.el8_2.x86_64 -> 5.9.0-rc7+

[LISAv2 Test Results Summary]
Test Run On           : 10/01/2020 07:43:35
ARM Image Under Test  : RedHat : RHEL : 8.2 : latest
Total Test Cases      : 7 (3 Passed, 4 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:3:9

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes)
-------------------------------------------------------------------------------------------------------------------------------------------
    1 POWER                *POWER-NETWORK-WORKLOAD-HIBERNATE                                                  FAIL                12.19
      ARMImageName: RedHat RHEL 8.2 latest, OverrideVMSize: Standard_D8s_v3, TestLocation: southcentralus, Kernel Version: 4.18.0-193.14.3.el8_2.x86_64
    2 POWER                POWER-HIBERNATE                                                                   PASS                53.61
      ARMImageName: RedHat RHEL 8.2 latest, OverrideVMSize: Standard_D8s_v3, TestLocation: southcentralus, Kernel Version: 4.18.0-193.14.3.el8_2.x86_64 -> 5.9.0-rc7+
    3 POWER                *POWER-STORAGE-WORKLOAD-HIBERNATE                                                  FAIL                 2.83
      ARMImageName: RedHat RHEL 8.2 latest, OverrideVMSize: Standard_D8s_v3, TestLocation: southcentralus, Kernel Version: 4.18.0-193.14.3.el8_2.x86_64 -> 5.9.0-rc7+
    4 POWER                *POWER-MEMORY-WORKLOAD-HIBERNATE                                                   FAIL                 8.67
      ARMImageName: RedHat RHEL 8.2 latest, OverrideVMSize: Standard_D8s_v3, TestLocation: southcentralus, Kernel Version: 4.18.0-193.14.3.el8_2.x86_64
    5 POWER                POWER-VMRESIZE-HIBERNATE                                                          PASS                46.28
      ARMImageName: RedHat RHEL 8.2 latest, OverrideVMSize: Standard_D8s_v3, TestLocation: southcentralus, Kernel Version: 4.18.0-193.14.3.el8_2.x86_64 -> 5.9.0-rc7+
    6 POWER                POWER-CLOCK-SYNC-HIBERNATE                                                        FAIL                 2.53
      ARMImageName: RedHat RHEL 8.2 latest, OverrideVMSize: Standard_D8s_v3, TestLocation: southcentralus, Kernel Version: 4.18.0-193.14.3.el8_2.x86_64 -> 5.9.0-rc7+
    7 POWER                POWER-HIBERNATE-SRIOV                                                             PASS                48.45
      ARMImageName: RedHat RHEL 8.2 latest, Networking: SRIOV, OverrideVMSize: Standard_D8s_v3, TestLocation: southcentralus, Kernel Version: 4.18.0-193.14.3.el8_2.x86_64 -> 5.9.0-rc7+

Found Azure disk attachment time varied. Sometimes took 42 seconds. Increased from 30 to 60 seconds waiting time. POWER-CLOCK-SYNC-HIBERNATE passed. 'stress-ng' is still broken - POWER-[STORAGE|NETWORK|MEMORY]-WORKLOAD-HIBERNATE are broken.

[LISAv2 Test Results Summary]
Test Run On           : 10/01/2020 18:01:18
ARM Image Under Test  : RedHat : RHEL : 8.2 : latest
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:1:2

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes)
-------------------------------------------------------------------------------------------------------------------------------------------
    1 POWER                POWER-CLOCK-SYNC-HIBERNATE                                                        PASS                59.85
      ARMImageName: RedHat RHEL 8.2 latest, OverrideVMSize: Standard_D8s_v3, TestLocation: southcentralus, Kernel Version: 4.18.0-193.14.3.el8_2.x86_64 -> 5.9.0-rc7+

### CODE REVIEW

**Review by dcui — DISMISSED** [review]
I had a quick look at this PR, and it looks good to me.

File: Testscripts/Linux/utils.sh [hunk 1]
```diff
@@ -3817,4 +3817,22 @@ function is_hpc_vm() {
...
-}
\ No newline at end of file
+}
+
+# Return the string from the dmesg, messages or syslog
+# Depending on distro, system logs are different.
+# if found, return 1. Otherwise, 0.
+function found_sys_log() {	
+	if [ -f /var/log/messages ]; then
+		ret=$(sudo cat /var/log/messages | grep -i "$1")
```
Comment (abhimarathe) [inline Testscripts/Linux/utils.sh]: should it be _ret ?

Comment (juhlee-microsoft) [reply Testscripts/Linux/utils.sh]: Fixed

File: Testscripts/Linux/utils.sh [hunk 2]
```diff
@@ -3817,4 +3817,22 @@ function is_hpc_vm() {
...
+# if found, return 1. Otherwise, 0.
+function found_sys_log() {
+	if [ -f /var/log/messages ]; then
+		_ret=$(sudo cat /var/log/messages | grep -i "$1")
+	elif [ -f /var/log/syslog ]; then
+		_ret=$(sudo cat /var/log/syslog | grep -i "$1")
+	else
+		_ret=$(sudo dmesg | grep -i "$1")
+	fi
+	if [[ "$_ret" == *"$1"* ]]; then
```
Comment (abhimarathe) [inline Testscripts/Linux/utils.sh:3833]: I think we can avoid this check, by depending on what the error code grep command returns. As we are doing same string matching 2 times with the same string.
Let me know your views on this.

**Review by abhimarathe — APPROVED** [review]

**Review by dcui — APPROVED** [review]
Looks good to me.

**Pull request merged**